src/game/logic/object/weapon.cpp

```diff
@@ -1195,15 +1370,236 @@ bool Weapon::Compute_Approach_Target(const Object *source_obj,
     float angle_offset,
     Coord3D &approach_target_pos) const
 {
-    // todo
-    return false;
+    const Coord3D *pos;
+    Coord3D vector;
+
+    if (target_obj != nullptr) {
+        pos = target_obj->Get_Position();
+        g_thePartitionManager->Get_Vector_To(target_obj, source_obj, FROM_BOUNDINGSPHERE_2D, vector);
+    } else if (target_pos != nullptr) {
+        pos = target_pos;
+        g_thePartitionManager->Get_Vector_To(source_obj, target_pos, FROM_BOUNDINGSPHERE_2D, vector);
+        vector.x = -vector.x;
+        vector.y = -vector.y;
+        vector.z = -vector.z;
```

Comment:
This direction is correct? target to source? Not the other way around?

Reply:
This one is also correct.